pkg/test/framework/components/istio/operator.go

```diff
@@ -400,18 +404,63 @@ spec:
 	return nil
 }
 
-func createCrossNetworkGateway(ctx resource.Context, cluster resource.Cluster, cfg Config) error {
+func (i *operatorComponent) createCrossNetworkGateway(ctx resource.Context, workDir string, cluster resource.Cluster, cfg Config) error {
 	scopes.Framework.Infof("Setting up cross-network-gateway in cluster: %s namespace: %s", cluster.Name(), cfg.SystemNamespace)
 
-	return ctx.Config(cluster).ApplyYAML(cfg.SystemNamespace, fmt.Sprintf(`
+	// generate input
+	gwYaml := fmt.Sprintf(`
+apiVersion: install.istio.io/v1alpha1
```

Comment:
Can we also update https://github.com/istio/istio/tree/master/samples/cross-network-gateway to do this same thing? Maybe a script that takes network name as an argument?